modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.server/src/main/scala/edu/gemini/seqexec/web/server/common/LogInitialization.scala

```diff
@@ -8,7 +8,26 @@ import scalaz._
 import Scalaz._
 import scalaz.effect.IO
 
-trait LogInitialization {
+trait AppBaseDir {
+  /**
+    * Calculates the base dir of the application based on the location of "this" class jar file
+    * It will throw an exception if unable to find the base dir
+    */
+  def baseDir: File = {
```

Comment:
It's public, so yes. `IO` actions should be small so we can compose them. Putting a big block of side-effecty stuff inside of `IO { ... }` doesn't buy us much.